api/src/api/resource.js

```diff
@@ -3,17 +3,98 @@ const router = express.Router();
 const { errorWrap } = require('../middleware');
 const Resource = require('../models/resource');
 
-// Get all resources (with query param "category")
+const addFields = {
+  $addFields: {
+    costSortId: {
+      $cond: [
+        { $eq: ['$cost', 'Free'] },
+        0,
+        {
+          $cond: [
+            { $eq: ['$cost', { $literal: '$' }] },
+            1,
+            {
+              $cond: [
+                { $eq: ['$cost', { $literal: '$$' }] },
+                2,
+                {
+                  $cond: [{ $eq: ['$cost', { $literal: '$$$' }] }, 3, 4],
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    },
+  },
+};
+
+// Get all resources (with query params)
 router.get(
   '/',
   errorWrap(async (req, res) => {
-    const { category } = req.query;
-    let resources;
-    if (category != null) {
-      resources = await Resource.find({ category: { $in: category } });
-    } else {
-      resources = await Resource.find();
+    const { category, subcategory, cost, language, city, sort } = req.query;
+
+    let query = {};
+    if (category != null && category !== '' && category !== 'All Resources') {
+      query = { category: category };
+      if (subcategory != null && subcategory !== '') {
+        query = { ...query, subcategory: subcategory };
+      }
+    }
+    if (cost != null) {
+      // get last part, if length is 1 then all resources
+      const parts = cost.split(' ');
+      if (parts.length > 0) {
+        if (parts.length === 1) {
+          // Free
+          query = { ...query, cost: 'Free' };
+        }
+        const price = parts[parts.length - 1];
+        if (price === '$$') {
+          query = {
+            ...query,
+            $or: [{ cost: price }, { cost: '$' }, { cost: 'Free' }],
+          };
+        } else if (price === '$') {
+          query = { ...query, $or: [{ cost: price }, { cost: 'Free' }] };
+        }
+        // If $$$, then it's all prices, so no need to query extra
+      }
+    }
+    if (language != null && language !== 'All') {
+      query = { ...query, availableLanguages: language };
     }
+    if (city != null && city !== 'All' && city !== 'All / Champaign County') {
+      const ignoreCase = new RegExp('^' + city, 'i');
+      query = { ...query, city: ignoreCase };
+    }
+
+    let orderBy = null;
+    if (sort != null) {
+      if (
+        sort === 'Cost' ||
```

Comment:
Is there any way to do this that isn't just hardcoding the languages?

Reply:
Not that I know of... I couldn't figure out a way to keep the "value" of sorting in english which is kinda sucky, so that's why I had to pass it through to the backend. Maybe to make it better I could keep this as an array constant?